cmd/template/cluster/provider/capa.go

```diff
@@ -120,6 +123,43 @@ func templateClusterAWS(ctx context.Context, k8sClient k8sclient.Interface, outp
 			},
 		}
 
+		if config.AWS.ClusterType == "proxy-private" {
+			c, _ := cidr.Parse(config.AWS.NetworkVPCCIDR)
+			subnets, err := c.SubNetting(cidr.MethodSubnetNum, 4)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			subnetCount := len(config.AWS.MachinePool.AZs)
+			if subnetCount == 0 {
+				subnetCount = config.AWS.NetworkAZUsageLimit
+			}
+
+			i := 0
+			for i < subnetCount {
+				flagValues.Network.Subnets = append(flagValues.Network.Subnets, capa.Subnet{
+					CidrBlock: subnets[i].CIDR().String(),
+				})
+				i++
+			}
+
+			httpProxy := config.AWS.HttpsProxy
+			if config.AWS.HttpProxy != "" {
+				httpProxy = config.AWS.HttpProxy
+			}
+			flagValues.Proxy = &capa.Proxy{
+				Enabled:    true,
+				HttpsProxy: config.AWS.HttpsProxy,
+				HttpProxy:  httpProxy,
+				NoProxy:    config.AWS.NoProxy,
+			}
+
+			flagValues.Network.APIMode = NetworkModePrivate
+			flagValues.Network.VPCMode = NetworkModePrivate
+			flagValues.Network.DnsMode = NetworkModePrivate
+			flagValues.Network.TopologyMode = TopoloyModeManaged
```

Comment:
we might wanna configurable the topology mode as for  some customers it has to be set to `UserManaged`